src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnectorConfig.java

```diff
@@ -0,0 +1,179 @@
+    return systemType;
+  }
+
+  /**
+   * Validates the credentials parameters.
+   *
+   * @param failureCollector {@code FailureCollector}
+   */
+  public void validateCredentials(FailureCollector failureCollector) {
+
+    if (AribaUtil.isNullOrEmpty(getClientId()) && !containsMacro(CLIENT_ID)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("Ariba Client Id");
+      failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(CLIENT_ID);
+    }
+    if (AribaUtil.isNullOrEmpty(getClientSecret()) && !containsMacro(CLIENT_SECRET)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("Ariba Client Secret");
+      failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(CLIENT_SECRET);
+    }
+    if (AribaUtil.isNullOrEmpty(getApiKey()) && !containsMacro(APIKEY)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("Ariba API Key");
+      failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(APIKEY);
+    }
+    if (AribaUtil.isNullOrEmpty(getBaseURL()) && !containsMacro(BASE_URL)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey(ResourceConstants.API_END_POINT);
+      failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(BASE_URL);
+    }
+    if (AribaUtil.isNullOrEmpty(getRealm()) && !containsMacro(REALM)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey(ResourceConstants.REALM_NAME);
+      failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(REALM);
+    }
+  }
+
+  public final void validateToken(FailureCollector collector) throws AribaException, IOException {
+    String tokenUrl = "https://api.au.cloud.ariba.com/v2/oauth/token";
+    HttpUrl.Builder builder = HttpUrl.parse(tokenUrl).newBuilder();
+    URL url = builder.build().url();
+    if (httpAribaTokenCall(url).code() != HttpURLConnection.HTTP_OK) {
+      collector.addFailure("Credentials are incorrect", "Please check the credentials");
+    }
+  }
+
+  Response httpAribaTokenCall(URL endpoint) throws IOException {
+    OkHttpClient enhancedOkHttpClient = new OkHttpClient();
+    Request req = buildTokenRequest(endpoint);
+    // No API limit on this call
+    return enhancedOkHttpClient.newCall(req).execute();
+  }
+
+  private Request buildTokenRequest(URL endpoint) {
+    String mediaType = MediaType.APPLICATION_FORM_URLENCODED;
+    RequestBody body = RequestBody.create(TOKEN_GRANT_TYPE,
+                                          okhttp3.MediaType.parse(mediaType));
+    return new Request.Builder()
+      .get().url(endpoint)
+      .method(POST, body)
+      .addHeader(AUTHORIZATION, String.format("Basic %s", getBase64EncodedValue()))
+      .addHeader(CONTENT_TYPE, mediaType)
+      .build();
+  }
+
+  /**
+   * call to get Base64 encoded value
+   *
+   * @return base64 String
+   */
+  private String getBase64EncodedValue() {
```

Comment:
This code also exists in AribaServices. Can we avoid duplication?

Reply:
Reused it.